irods-vfs-impl/src/main/java/org/irods/nfsrods/vfs/IRODSVirtualFileSystem.java

```diff
@@ -1645,15 +1663,101 @@ private static Inode toFh(long _inodeNumber)
         return Inode.forFile(Longs.toByteArray(_inodeNumber));
     }
 
+    private String tryAndFindPathFromGenQuery(long _inodeNumber, String option) {
```

Comment:
Consider renaming to something like `findLogicalPathByInodeNumber`.

That's just an example. Let the method name, return type, and parameters help you express intent and purpose.

GenQuery is an implementation detail.